main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2107,7 +2107,11 @@ object Defaults extends BuildCommon {
           val projectName = name.value
           if (ScalaArtifacts.isScala3(sv)) {
             val project = if (config == Compile) projectName else s"$projectName-$config"
-            Seq("-project", project)
+            if (scalaVersion.value.startsWith("3.0.0")) {
+              Seq("-project", project)
```

Comment:
We could still pass -scalajs here to avoid most issues without introducing warnings:
```suggestion
              compileOptions.filter(_ == "-scalajs") ++ Seq("-project", project)
```

Reply:
Is `scaladoc` 3.0.0 able to use `-scalajs`? I think it was failing in `cats-effect` repo even with `-scalajs`.

Reply:
That's true. We didn't handle `-scalajs` in 3.0.0.